booster/library/Booster/Pattern/ApplyEquations.hs

```diff
@@ -817,17 +818,46 @@ applyEquation term rule =
                         -- could now be syntactically present in the path constraints, filter again
                         stillUnclear <- lift $ filterOutKnownConstraints knownPredicates unclearConditions
 
-                        -- abort if any of the conditions is still unclear at that point
+                        mbSolver :: Maybe SMT.SMTContext <- (.smtSolver) <$> lift getConfig
+
+                        -- check any conditions that are still unclear with the SMT solver
+                        -- (or abort if no solver is being used), abort if still unclear after
                         unless (null stillUnclear) $
-                            throwE
-                                ( \ctxt ->
-                                    ctxt $
-                                        logMessage $
-                                            renderOneLineText $
-                                                "Uncertain about a condition(s) in rule:"
-                                                    <+> hsep (intersperse "," $ map (pretty' @mods) unclearConditions)
-                                , IndeterminateCondition unclearConditions
-                                )
+                            let checkWithSmt :: SMT.SMTContext -> EquationT io (Maybe Bool)
+                                checkWithSmt smt =
+                                    SMT.checkPredicates smt knownPredicates mempty (Set.fromList stillUnclear) >>= \case
+                                        Left SMT.SMTSolverUnknown{} -> do
+                                            -- FIXME logging?
+                                            pure Nothing
+                                        Left _other ->
+                                            pure Nothing -- liftIO $ Exception.throw other
```

Comment:
```suggestion
                                            pure Nothing
```

Reply:
Maybe we should indeed fail hard here? (this is an actual SMT _error_)